resources/helm/v2.0/istio-control/istio-discovery/templates/ior-roles.yaml

```diff
@@ -0,0 +1,35 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: istiod-ior
+  namespace: {{ .Release.Namespace }}
+  labels:
+    maistra-version: 2.0.0
+    app: istiod
+    release: {{ .Release.Name }}
+rules:
+- apiGroups: ["route.openshift.io"]
+  resources: ["routes", "routes/custom-host"]
+  verbs: ["get", "list", "watch", "create", "delete"]
+- apiGroups: [""]
+  resources: ["services", "pods"]
+  verbs: ["get", "list"]
```

Comment:
Can we just add these rules to existing clusterrole.yaml?

Reply:
@rcernich  These are just roles, not cluster roles. Also, having them in a separate files means one less `sed` hacking.

Reply:
There's a difference between a role and a role binding.  We use cluster roles because they can be referenced by role bindings in other namespaces.  When a namespace is configured to be a part of the mesh, we create role bindings for istio (galley, pilot, etc.) which reference the cluster role.  If we used namespaced roles, we'd have to create the roles in each namespace as well.

As far as being separate, it just means we need to create an extra role binding too.

Reply:
@rcernich done.